pkgs/collection/lib/src/iterable_extensions.dart

```diff
@@ -613,6 +613,33 @@ extension IterableExtension<T> on Iterable<T> {
       yield slice;
     }
   }
+
+  /// Lazily returns a new iterable where a [separator] is placed between each
+  /// element of this iterable.
+  ///
+  /// This is the most performant way to implement this functionality because
+  /// it uses a generator (`sync*`) and does not create a new list upfront,
+  /// making it memory-efficient for iterables of any size.
+  ///
+  /// Example:
+  /// ```dart
+  /// final numbers = [1, 2, 3];
+  /// final withSeparators = numbers.intersperse(0);
+  /// print(withSeparators.toList()); // Output: [1, 0, 2, 0, 3]
+  /// ```
+  ///
+  /// The resulting iterable is lazy. It only computes the next value when
+  /// it's requested.
```

Comment:
I wouldn't say this. That's what iterables are expected to do.
A function returning an `Iterable` should document if it *doesn't* work that way.